actionBar.js

```diff
@@ -0,0 +1,103 @@
+const St = imports.gi.St;
+const PopupMenu = imports.ui.popupMenu;
+const GObject = imports.gi.GObject;
+const Clutter = imports.gi.Clutter;
+const Lang = imports.lang;
+
+const ExtensionUtils = imports.misc.extensionUtils;
+const Me = ExtensionUtils.getCurrentExtension();
+const ConfirmDialog = Me.imports.confirmDialog;
+const Utils = Me.imports.utils;
+
+const Gettext = imports.gettext.domain("clipboard-indicator");
+const _ = Gettext.gettext;
+
+var ActionBar = GObject.registerClass(
+  class ActionBar extends PopupMenu.PopupBaseMenuItem {
+    _init() {
+      super._init({
+        activate: false,
+        hover: false,
+        style_class: 'ci-action-bar-box',
+      })
+
+      this.actionsBox = new St.BoxLayout({
+        vertical: false,
+        style_class: 'ci-action-bar-box-layout',
+      });
+
+      this.actionsBox.set_x_expand(true);
+      this.actionsBox.set_y_expand(true);
+
+      // TODO:: Add tooltip
+      // Private mode switch
+      this.privateModeBtn = new PopupMenu.PopupSwitchMenuItem(
+        _("Private mode"), false, {
+        reactive: true, hover: true,
+      });
+      this.privateModeBtn.set_x_expand(true);
+      this.privateModeBtn.set_y_expand(true);
+      this.actionsBox.add(this.privateModeBtn);
```

Comment:
Maybe use a spacer to avoid having the switch closer to the delete-all icon than the label ?
```suggestion
      this.actionsBox.add(this.privateModeBtn);
      // Add a spacer
      this.spacer = new PopupMenu.PopupBaseMenuItem();
      this.spacer.set_x_expand(true);
      this.spacer.set_y_expand(true);
      this.actionsBox.add(this.spacer);      
```

Reply:
Damned, the spacer is reacting on hover by default ... we can make it silent :
```suggestion
      this.actionsBox.add(this.privateModeBtn);

      // Add a spacer
      this.spacer = new PopupMenu.PopupBaseMenuItem({reactive: false, hover: false});
      this.spacer.set_x_expand(true);
      this.spacer.set_y_expand(true);
      this.actionsBox.add(this.spacer);
```